arrow-rs/data/src/records.rs

```diff
@@ -97,7 +97,7 @@ impl TryFrom<LegacyRecords> for SchemaChunk<SchemaRef> {
             .map_err(|_| ChunkError::LengthMismatch)?;
 
         Ok(Self {
-            schema,
+            schema: Schema::clone(&schema),
```

Comment:
This may be semantically a little bit cleaner.
```suggestion
            schema: schema.unwrap_or_clone(),
```